aws/resource_aws_athena_named_query.go

```diff
@@ -0,0 +1,87 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/athena"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsAthenaNamedQuery() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAthenaNamedQueryCreate,
+		Read:   resourceAwsAthenaNamedQueryRead,
+		Delete: resourceAwsAthenaNamedQueryDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"query": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"database": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAthenaNamedQueryCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).athenaconn
+
+	input := &athena.CreateNamedQueryInput{
+		Database:    aws.String(d.Get("database").(string)),
+		Name:        aws.String(d.Get("name").(string)),
+		QueryString: aws.String(d.Get("query").(string)),
+	}
+	if raw, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(raw.(string))
+	}
+
+	resp, err := conn.CreateNamedQuery(input)
+	if err != nil {
+		return err
+	}
+	d.SetId(*resp.NamedQueryId)
+	return resourceAwsAthenaNamedQueryRead(d, meta)
+}
+
+func resourceAwsAthenaNamedQueryRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).athenaconn
+
+	input := &athena.GetNamedQueryInput{
+		NamedQueryId: aws.String(d.Id()),
+	}
+
+	_, err := conn.GetNamedQuery(input)
+	if err != nil {
+		return err
```

Comment:
Do you mind adding a check here for cases when the query is gone and remove it from the state via `d.SetId("")`, please? I don't know if it returns an error or just `nil` in such case (that may need some research), but we should have a check in place.